sotodlib/site_pipeline/make_ml_map.py

```diff
@@ -127,189 +150,203 @@ def main(config_file=None, defaults=defaults, **args):
+                    good_dets = mapmaking.find_usable_detectors(obs, glitch_flags="glitch_flags")
+                    obs.restrict("dets", good_dets)
+                    if obs.dets.count == 0:
+                        logger.debug(f"Skipped {name} (all dets cut)")
+                        continue
+                    # Gapfill glitches. This function name isn't the clearest
+                    tod_ops.get_gap_fill(obs, flags=obs.flags.glitch_flags, swap=True)
+                    # Gain calibration
+                    gain  = 1
+                    for gtype in ["relcal", "abscal"]:
+                        gain *= obs[gtype][:, None]
+                    obs.signal *= gain
+                    # Fourier-space calibration
+                    fsig = fft.rfft(obs.signal)
+                    freq = fft.rfftfreq(obs.samps.count, 1/srate)
+                    # iir filter
+                    iir_filter = tod_ops.filters.iir_filter()(freq, obs)
+                    fsig /= iir_filter
+                    gain /= iir_filter[0].real # keep track of total gain for our record
+                    fsig /= tod_ops.filters.timeconst_filter(None)(freq, obs)
+                    fft.irfft(fsig, obs.signal, normalize=True)
+                    del fsig
+
+                    # Apply pointing correction.
+                    #obs.focal_plane.xi    += obs.boresight_offset.xi
+                    #obs.focal_plane.eta   += obs.boresight_offset.eta
+                    #obs.focal_plane.gamma += obs.boresight_offset.gamma
+                    obs.focal_plane.xi += obs.boresight_offset.dx
+                    obs.focal_plane.eta += obs.boresight_offset.dy
+                    obs.focal_plane.gamma += obs.boresight_offset.gamma
+
+                # Injecting at this point makes us insensitive to any bias introduced
+                # in the earlier steps (mainly from gapfilling). The alternative is
+                # to inject it earlier, and then anti-calibrate it
+                if args['inject']:
+                    mapmaking.inject_map(obs, map_to_inject, recenter=recenter, interpol=args['interpol'])
+                utils.deslope(obs.signal, w=5, inplace=True)
+
+                if args['downsample'] != 1:
+                    obs = mapmaking.downsample_obs(obs, pass_cfg['downsample'])
+
+                # Maybe load precomputed noise model
+                nmat_file = f"{nmat_dir}/nmat_{name}.hdf"
+                if args['nmat_mode'] == "load" or args['nmat_mode'] == "cache" and os.path.isfile(nmat_file):
+                    logger.info(f"Reading noise model {nmat_file}")
+                    nmat = mapmaking.read_nmat(nmat_file)
+                else: nmat = None
+
+                # And add it to the mapmaker
+                # FIXME: How to handle multipass here?
+                with mapmaking.mark(f"add_obs {name}"):
+                    signal_estimate = None if pass_ind == 0 else mapmaker.transeval(name, obs, mapmaker_prev, x_prev)
```

Comment:
I tested the mapmaker with 2 passes. Once the second pass started, it failed in this line like this
```
Traceback (most recent call last):
  File "/pscratch/sd/c/chervias/cmbenv-perlmutter-so2/bin/so-site-pipeline", line 8, in <module>
    sys.exit(main())
  File "/pscratch/sd/c/chervias/cmbenv-perlmutter-so2/lib/python3.10/site-packages/sotodlib/site_pipeline/cli.py", line 109, in main
    module.main(**vars(args))
  File "/pscratch/sd/c/chervias/cmbenv-perlmutter-so2/lib/python3.10/site-packages/sotodlib/site_pipeline/make_ml_map.py", line 297, in main
    signal_estimate = None if pass_ind == 0 else mapmaker.transeval(name, obs, mapmaker_prev, x_prev)
  File "/pscratch/sd/c/chervias/cmbenv-perlmutter-so2/lib/python3.10/site-packages/sotodlib/mapmaking/ml_mapmaker.py", line 143, in transeval
    ssig.transeval(id, obs, osig, oval, tod=tod)
  File "/pscratch/sd/c/chervias/cmbenv-perlmutter-so2/lib/python3.10/site-packages/sotodlib/mapmaking/ml_mapmaker.py", line 359, in transeval
    pmap.from_map(dest=tod, signal_map=map, comps=self.comps)
  File "/pscratch/sd/c/chervias/cmbenv-perlmutter-so2/lib/python3.10/site-packages/sotodlib/coords/pmat.py", line 451, in from_map
    proj.from_map(self._prepare_map(signal_map), self._get_asm(), signal=dest, comps=comps)
  File "/pscratch/sd/c/chervias/cmbenv-perlmutter-so2/sotoast2/lib/python3.10/site-packages/so3g/proj/wcs.py", line 679, in from_map
    return super().from_map(src_map, assembly, signal, comps)
  File "/pscratch/sd/c/chervias/cmbenv-perlmutter-so2/sotoast2/lib/python3.10/site-packages/so3g/proj/wcs.py", line 321, in from_map
    signal_out = projeng.from_map(
RuntimeError: Tiling problem (index 11): Projector expects tile but it is missing.
```

Reply:
@chervias this is the same issue that is holding #947. @mhasself and I have tried to resolve it, but we need to schedule another debug session.

Can you paste the test you run? If we can run we may be able to replace it and see something new compared to my tests.